File: apps/sdkwork-appstore-flutter-mobile/packages/sdkwork_appstore_flutter_mobile_search/lib/src/services/search_service.dart
import 'package:sdkwork_appstore_flutter_mobile_core/sdkwork_appstore_flutter_mobile_core.dart';

import '../models/search_models.dart';

/// Search service.
///
/// The App Store app SDK clients are injected by the root bootstrap
/// (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
/// never issues raw HTTP.
class SearchService {
  const SearchService({required this.clients});

  final AppstoreAppSdkClients clients;

  String get capability => 'search';

  SearchPageResult<Never> empty() =>
      const SearchPageResult<Never>(items: <Never>[]);
}

File: apps/sdkwork-appstore-mini-program/packages/sdkwork-appstore-mp-search/src/services/searchService.ts
import type { AppstoreAppClient } from "@sdkwork/appstore-mp-core";

import type { SearchPageResult } from "../types/searchModels.js";

/**
 * Search service.
 *
 * The generated App Store app SDK client is injected by the root bootstrap
 * (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
 * never issues raw HTTP.
 */
export interface SearchService {
  readonly capability: "search";
}

export function createSearchService(client: AppstoreAppClient): SearchService {
  void client;
  return { capability: "search" };
}

export function emptySearchResult(): SearchPageResult<never> {
  return { items: [] };
}
